querybook/server/logic/vector_store.py

```diff
@@ -43,68 +40,97 @@ def _get_query_doc_id(query: str) -> str:
     return f"query_{generate_hash(query)}"
 
 
+def get_sample_query_cells_by_table_name(table_name: str, k: int = 50):
+    """Get at most 50 sample queries from elasticsearch query_cell index by table name in the past half year."""
+    from lib.elasticsearch.search_query import construct_query_search_query
+    from lib.elasticsearch.search_utils import get_matching_objects
+    from logic.elasticsearch import ES_CONFIG
+
+    # get timestamp of yesterday
+    end_time = int(time.time()) - 24 * 60 * 60
+    # get timestamp of 6 months ago
+    start_time = end_time - 6 * 30 * 24 * 60 * 60
+
+    filters = [
+        ["full_table_name", [f"{table_name}"]],
+        ["query_type", "query_cell"],
+        ["statement_type", ["SELECT"]],
+        ["startDate", start_time],
+        ["endDate", end_time],
+    ]
+
+    query = construct_query_search_query(
+        uid=current_user.id if current_user else 1,
+        keywords="",
+        filters=filters,
+        limit=min(
+            3 * k, 50
```

Comment:
maybe the 50 should be in a constant, there are a bit too many constant numbers flying around